test/e2e/gateway_api_test.go

```diff
@@ -109,54 +108,62 @@ func testGatewayAPIResources(t *testing.T) {
+// successfully:
+//
+//   - The required Subscription and CatalogSource are created.
+//
+//   - The Sail operator is installed successfully, and it reports status Running
+//     and Ready.
+//
+//   - Istiod is installed successfully and has status Running and Ready.
+//
+//   - The Istio CR is created successfully.
+//
+//   - If the Istio and Subscription CRs are deleted, they are recreated
+//     automatically.
 func testGatewayAPIIstioInstallation(t *testing.T) {
-	t.Helper()
-
+	t.Log("Checking for the Subscription...")
 	if err := assertSubscription(t, openshiftOperatorsNamespace, expectedSubscriptionName); err != nil {
 		t.Fatalf("failed to find expected Subscription %s: %v", expectedSubscriptionName, err)
 	}
+	t.Log("Checking for the CatalogSource...")
 	if err := assertCatalogSource(t, expectedCatalogSourceNamespace, expectedCatalogSourceName); err != nil {
 		t.Fatalf("failed to find expected CatalogSource %s: %v", expectedCatalogSourceName, err)
 	}
+	t.Log("Checking for the Sail operator deployment and pods...")
 	if err := assertOSSMOperator(t); err != nil {
 		t.Fatalf("failed to find expected Istio operator: %v", err)
 	}
+	t.Log("Checking for the Istiod pods...")
 	if err := assertIstiodControlPlane(t); err != nil {
 		t.Fatalf("failed to find expected Istiod control plane: %v", err)
 	}
-	// TODO - In OSSM 3.x the configuration object to check will be different.
-	if err := assertSMCP(t); err != nil {
-		t.Fatalf("failed to find expected SMCP: %v", err)
+	t.Log("Checking for the Istio CR...")
+	if err := assertIstio(t); err != nil {
+		t.Fatalf("failed to find expected Istio: %v", err)
 	}
-	// delete existing SMCP to test it gets recreated
-	if err := deleteExistingSMCP(t); err != nil {
-		t.Fatalf("failed to delete existing SMCP: %v", err)
+	t.Log("Deleting the Istio CR...")
+	if err := deleteExistingIstio(t); err != nil {
+		t.Fatalf("failed to delete existing Istio: %v", err)
 	}
-	// check if SMCP gets recreated
-	if err := assertSMCP(t); err != nil {
-		t.Fatalf("failed to find expected SMCP: %v", err)
+	t.Log("Checking that the Istio CR gets recreated...")
+	if err := assertIstio(t); err != nil {
+		t.Fatalf("failed to find expected Istio: %v", err)
 	}
-	// delete existing Subscription to test it gets recreated
+	t.Log("Deleting the Subscription...")
 	if err := deleteExistingSubscription(t, openshiftOperatorsNamespace, expectedSubscriptionName); err != nil {
 		t.Fatalf("failed to delete existing Subscription %s: %v", expectedSubscriptionName, err)
 	}
-	// checks if subscription gets recreated.
+	t.Log("Checking that the Subscription gets recreated...")
 	if err := assertSubscription(t, openshiftOperatorsNamespace, expectedSubscriptionName); err != nil {
 		t.Fatalf("failed to find expected Subscription %s: %v", expectedSubscriptionName, err)
 	}
 }
 
 // testGatewayAPIObjects tests that Gateway API objects can be created successfully.
 func testGatewayAPIObjects(t *testing.T) {
-	t.Helper()
```

Comment:
You could remove the one on L188 too?

Reply:
Yup, `testGatewayAPIResourcesProtection` didn't exist when I made this change.  Fixed in https://github.com/openshift/cluster-ingress-operator/compare/30b576c4293b2962b137bcb93ff77f278d7bf681..26a3918c7374cc16eac04b87c9e4267591c304a5.